sys/net/gnrc/sock/ip/gnrc_sock_ip.c

```diff
@@ -126,14 +129,22 @@ ssize_t sock_ip_recv_buf(sock_ip_t *sock, void **data, void **buf_ctx,
         return -EADDRNOTAVAIL;
     }
     tmp.family = sock->local.family;
-    res = gnrc_sock_recv((gnrc_sock_reg_t *)sock, &pkt, timeout, &tmp);
+    sock_ip_ep_t *local = NULL;
+#ifdef MODULE_SOCK_AUX_LOCAL
+    local = (aux != NULL) ? &aux->local : NULL;
+#endif
+    res = gnrc_sock_recv((gnrc_sock_reg_t *)sock, &pkt, timeout, &tmp, local);
     if (res < 0) {
         return res;
     }
     if (remote != NULL) {
         /* return remote to possibly block if wrong remote */
         memcpy(remote, &tmp, sizeof(tmp));
     }
+    if (aux != NULL) {
+        if (IS_USED(MODULE_SOCK_AUX_LOCAL))
```

Comment:
Nit: This would read more consistently with the above check if the same form (ifdef or if (IS_USED(...))) were used.

Reply:
`if (IS_USED(…))` is preferred (giving the optimizer precedence over the CPP), but if there is no way around using `#if IS_USED(…)` (e.g. certain struct fields as above), there is no way around this, so I would prefer it to keep "as is", since that reflects the style guides. However, it should really be `#if IS_USED(…)` above.